### PR TITLE
[Process] Return built-in cmd.exe commands directly in ExecutableFinder

### PR DESCRIPTION
**Pull request by Seldaek**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Refs #58734
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Process/ExecutableFinder.php [hunk 1]
```diff
@@ -48,6 +55,11 @@ public function addSuffix(string $suffix)
      */
     public function find(string $name, ?string $default = null, array $extraDirs = [])
     {
+        // windows built-in commands that are present in cmd.exe should not be resolved using PATH as they do not exist as exes
+        if ('\\' === \DIRECTORY_SEPARATOR && in_array(strtolower($name), self::CMD_BUILTIN_COMMANDS, true)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/ExecutableFinder.php]:
```suggestion
        if ('\\' === \DIRECTORY_SEPARATOR && \in_array(strtolower($name), self::CMD_BUILTINS, true)) {
```

**Review by nicolas-grekas — APPROVED** [review]
(with minor CS change)

**Comment by nicolas-grekas** [comment]
Thank you @Seldaek.

File: src/Symfony/Component/Process/Tests/ExecutableFinderTest.php [hunk 2]
```diff
@@ -159,6 +159,18 @@ public function testEmptyDirInPath()
...
     }
 
+    public function testFindBuiltInCommandOnWindows()
+    {
+        if ('\\' !== \DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('Can be only tested on windows');
+        }
+
+        $finder = new ExecutableFinder();
+        $this->assertSame('rmdir', $finder->find('RMDIR'));
```
Comment (xabbuh) [inline src/Symfony/Component/Process/Tests/ExecutableFinderTest.php:169]: AppVeyor returns `RMDIR` but not `rmdir` (see the failing test). Do we need to fix the expectation here or is there still some flaw in the code?

Comment (Seldaek) [reply src/Symfony/Component/Process/Tests/ExecutableFinderTest.php:169]: the command names are case insensitive so I guess the implementation is fine but test expectations should be fixed to match the input's case.

Comment (xabbuh) [reply src/Symfony/Component/Process/Tests/ExecutableFinderTest.php:169]: 👍 thanks for the feedback (see #58748 then)

**Pull request merged**